module-2/backend/week_two.md

```diff
@@ -1,37 +1,93 @@
 
-Note: When you're done, submit a PR. 
+Note: When you're done, submit a PR.
 
 1. At a high level, what is ActiveRecord? What does it do/allow you to do?
+ * ActiveRecord is a tool that works with relational databases that allows you to interact and query your database in specific ways.
+
 2. What kind of methods are `belongs_to`, and `has_many`? (i.e. class or instance) Give an example.
+ * they are class methods. For example a payload belongs_to a url but a url has_many payloads.
+
 3. What do they allow you to do?
+ * They allow you to add a class method that you can call on a single object. You can call Url.find(1).payloads (which will give you all the payload objects that belong to that specific Url)
+
 4. What's the difference between agile workflow and waterfall method?
+ * Agile workflow enables you to work in chunks as well as adapting/changing your work on the fly. A waterfall methods gives you a defined set of rules to follow which allows for less change and adaptation.
+
 5. What is the difference between `#new` and `#create`?
+ * New uses a get verb that lets you read the form where as create uses a post verb that send new data back to your database
+
 6. At a basic level, what does cURL allow you to do?
+ * A curl allows you to send data requests to your server
+
 7. In a database that's holding students and teachers, what will be the relationship between students and teachers? Draw the schema diagram.
+ * A think the relationship will be a many to many relationship which means you have to create a join table (TeacherStudents)  with a foreign key of student_id and  Teacher_id. Students can access there teachers through TeachersStudents and Teachers can access there students through TeachersStudents. ![Alt text](students.png)  
+
 8. Define foreign key, primary key, and schema.
+ * A foreign is usually seen as a example_id which links to another table that holds the information. Foreign keys are a way to make relationships between class objets. A Primary key does not need to be normalized or abstracted out to there own table. A schema is the blueprint of your database that holds all the information about your tables.
+
 9. Describe the relationship between a foreign key on one table and a primary key on another table.
+ * A foreign key on a table like payload which had request_type_id would link to a separate request_type table. In that request type table it would contain a primary key of http_verb. If you expect to get a lot of repeated data often times it is best to create a foreign key that bring you to a new table.
+
 10. What are the parts of an HTTP response?
-11. Describe some techniques to make our Sinatra code more DRY. Give an example of when you would use these techniques.
+ * hmm? response the serves gives back to the client
```

Comment:
The response is made up of headers and a body. The first header includes the status code. Those are the big pieces we'll be concerned about for now.